NuRadioReco/modules/LOFAR/pipelineVisualizer_LOFAR.py

```diff
@@ -0,0 +1,397 @@
+                        else:
+                            good_channel_pair_ids[ind, 1] = channel.get_id()
+
+                    # Check if dominant channel has been flagged
+                    channel = station.get_channel(good_channel_pair_ids[ind, 0])
+                    if channel.get_id() not in flagged_channels:
+                        good_antennas_dict[station.get_id()].append(channel.get_id())
+
+        fig_time, ax = plt.subplots(dpi=150, figsize=(8, 5))
+
+        triggered_station_ids = [station.get_id() for station in event.get_stations() if station.get_parameter(stationParameters.triggered)]    
+        num_stations = len(triggered_station_ids)
+        cmap = get_cmap('jet')
+        norm = Normalize(vmin=0, vmax=num_stations-1) 
+        
+        fluences = []
+        positions = []
+        SNRs = []
+        for station in event.get_stations():
+            if station.get_parameter(stationParameters.triggered):
+                zenith = station.get_parameter(stationParameters.cr_zenith)
+                azimuth = station.get_parameter(stationParameters.cr_azimuth)
+                good_antennas = good_antennas_dict[station.get_id()]
+                if len(good_antennas) >= min_number_good_antennas:
+                    for antenna in good_antennas:
+                        positions.append(detector.get_relative_position(station.get_id(), antenna) + detector.get_absolute_position(station.get_id()))
+                        channel = station.get_channel(antenna)
+                        SNRs.append(channel.get_parameter(channelParameters.SNR))
+                        fluences.append(np.sum(np.square(channel.get_trace())))
+                    station_pos = detector.get_absolute_position(station.get_id())
+                    ax.quiver(station_pos[0], station_pos[1], 
+                            np.cos(azimuth), np.sin(azimuth), 
+                            color='black', 
+                            scale=0.02, 
+                            scale_units='xy', 
+                            angles='uv',
+                            width=0.005)
+        
+        timelags = []
+        for station in event.get_stations():
+            if station.get_parameter(stationParameters.triggered):
+                good_antennas = good_antennas_dict[station.get_id()]
+                if len(good_antennas) >= min_number_good_antennas:
+                    for channel_id in good_antennas:
+                        timelags.append(station.get_channel(channel_id).get_parameter(channelParameters.signal_time))
+
+        for i, station in enumerate(event.get_stations()):
+            # plot absolute station positions
+            if station.get_parameter(stationParameters.triggered):
+                station_pos = detector.get_absolute_position(station.get_id())
+                ax.scatter(station_pos[0], station_pos[1], color=cmap(norm(i)), s=20, label=f'Station CS{station.get_id():03d}')
```

Comment:
maybe a different marker would be good to differentiate between station positions and antenna positions later (square or star?)

Reply:
good point! (pun intended)

Reply:
don't use points!